configs/memcached/cluster/topology.yaml

```diff
@@ -0,0 +1,27 @@
+kind: memcached
```

Comment:
Now we only have the kind `curvebs` and `curvefs`, the kind is designed for top-level of curve system. If we have object storage in the future, we can add `curveos` kind, besides that, i don't think we should add any kind.
And as user, they want to deploy curve as simple as possible, if we provide too many configure files, it becomes a burden for them.
I think the `memcache` is the part of curvefs system or related to curvefs system, you can deploy it like `snapshotclone` service and if you want to control deploy action, you can provide all kind of command option.

Reply:
The memcached cluster should be a separate cluster from curvefs. Since it is a cache cluster for client, different fs uses different memcached clusters. The difference from snapshotclone is that the deployment and destruction of memcached clusters are more flexible, and sometimes multiple sets of memcached clusters need to be deployed. In this way, the topology of curvefs must be modified for each deployment and destruction, and a destructive operation may affect the entire cluster. cause harm. I agree with you that the deployment system supported by curveadm should be kept as simple as possible. Too many configuration files will cause a burden on users, but I think this should be as consistent as possible through the design of configuration files and reduce the need for users to modify files to achieve.

Reply:
> sometimes multiple sets of memcached clusters need to be deployed

It means some clients sharing one memcache cluster and someone sharing another one? is it right?

If so i recommend add a management command to manage memcache cluster, like `pfs` and `target`, please don't mix it into curve cluster related command (`deploy`,`stop`,`clean` and etc...)，and there are many functions you can reuse for memcache management command.

> I think this should be as consistent as possible through the design of configuration files and reduce the need for users to modify files to achieve

I agree! you can use configure file to specify the topology of memcache cluster, and if you decide to use management command please remove the kind field.

BTW: why we don't use curveadm `playbook` to do this?

Reply:
> It means some clients sharing one memcache cluster and someone sharing another one? is it right?
yes

> BTW: why we don't use curveadm playbook to do this?
Does palybook only support deployment on one machine? memcached can be deployed on multiple machines.

Reply:
> Does palybook only support deployment on one machine? memcached can be deployed on multiple machines.

The `playbook` is a bit like `Ansible`, it allow you write bash script and execute in multi machines, it's easy to use and flexible, and i think it's more transparent and simpler for user, because all operations written in script.
It designed for deploy some components, maintain machines and etc...

Reply:
> The `playbook` is a bit like `Ansible`, it allow you write bash script and execute in multi machines, it's easy to use and flexible, and i think it's more transparent and simpler for user, because all operations written in script. It designed for deploy some components, maintain machines and etc...

How does palybook manage the memcached cluster? For example, if multiple memcached cluster services are deployed on the same machine, if you want to destroy only one memcached cluster service, you need to record the docker containers started by each cluster on each machine. This operation is no different from deploy .